File: Daily Challanges/2026-05-May/2026-05-01.py
from collections import defaultdict
def group_anagrams(words):
    groups = defaultdict(list)
    for word in words:
        sorted_word = ''.join(sorted(word))
        groups[sorted_word].append(word)
    return list(groups.values())

print(group_anagrams(["listen", "silent", "hello", "enlist", "world"]))
print(group_anagrams(["eat", "tea", "tan", "ate", "nat", "bat"]))

File: Daily Challanges/2026-05-May/2026-05-01.js
function groupAnagrams(words) {
  const groups = {};
  for (const word of words) {
    const count = new Array(26).fill(0);
    for (let i = 0; i < word.length; i++) {
      count[word.charCodeAt(i) - 97]++;
    }
    const key = count.join("#");
    if (!groups[key]) {
      groups[key] = [];
    }
    groups[key].push(word);
  }
  return Object.values(groups);
}
console.log(groupAnagrams(["listen", "silent", "hello", "enlist", "world"]));
console.log(groupAnagrams(["eat", "tea", "tan", "ate", "nat", "bat"]));
